quantities/tuples.hpp

```diff
@@ -5,10 +5,23 @@
 #include <tuple>
 #include <utility>
 
+#include "base/not_constructible.hpp"
+
 namespace principia {
 namespace quantities {
 namespace internal_tuples {
 
+using base::not_constructible;
+
+// A trait for finding if something is a tuple.
```

Comment:
The property we care about is probably is_cartesian_product, which is true for tuple, array, and pair. Let's bikeshed that another day, but add a todo.

Reply:
Done.